src/ray/core_worker/core_worker.h

```diff
@@ -2,17 +2,58 @@
 #define RAY_CORE_WORKER_CORE_WORKER_H
 
 #include "ray/common/buffer.h"
+#include "ray/core_worker/actor_handle.h"
 #include "ray/core_worker/common.h"
 #include "ray/core_worker/context.h"
 #include "ray/core_worker/object_interface.h"
 #include "ray/core_worker/profiling.h"
 #include "ray/core_worker/task_execution.h"
-#include "ray/core_worker/task_interface.h"
+#include "ray/core_worker/transport/direct_actor_transport.h"
 #include "ray/gcs/redis_gcs_client.h"
 #include "ray/raylet/raylet_client.h"
 
 namespace ray {
 
+/// Options of a non-actor-creation task.
+struct TaskOptions {
+  TaskOptions() {}
+  TaskOptions(int num_returns, std::unordered_map<std::string, double> &resources)
+      : num_returns(num_returns), resources(resources) {}
+
+  /// Number of returns of this task.
+  int num_returns = 1;
+  /// Resources required by this task.
+  std::unordered_map<std::string, double> resources;
+};
+
+/// Options of an actor creation task.
+struct ActorCreationOptions {
+  ActorCreationOptions() {}
+  ActorCreationOptions(uint64_t max_reconstructions, bool is_direct_call,
+                       const std::unordered_map<std::string, double> &resources,
+                       const std::unordered_map<std::string, double> &placement_resources,
+                       const std::vector<std::string> &dynamic_worker_options)
+      : max_reconstructions(max_reconstructions),
+        is_direct_call(is_direct_call),
+        resources(resources),
+        placement_resources(placement_resources),
+        dynamic_worker_options(dynamic_worker_options) {}
+
+  /// Maximum number of times that the actor should be reconstructed when it dies
+  /// unexpectedly. It must be non-negative. If it's 0, the actor won't be reconstructed.
+  const uint64_t max_reconstructions = 0;
```

Comment:
Is this thing actually used?

Reply:
I think it's used in some testing code. I'm just moving it over from a different file in this PR.